src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.hellbender.tools.funcotator;
+
+import htsjdk.tribble.Feature;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+
+import java.util.List;
+
+/**
+ * An abstract class to allow for the creation of a {@link Funcotation} for a given data source.
+ * Created by jonn on 8/30/17.
+ */
+public abstract class DataSourceFuncotationFactory implements AutoCloseable {
+
+    /**
+     * Perform cleanup tasks for this {@link DataSourceFuncotationFactory}.
+     */
+    public void close() {}
+
+    /**
+     * @return An ordered list of the names of annotations that this Data Source supports.
+     */
+    public abstract List<String> getSupportedFuncotationFields();
+
+    /**
+     * Creates a {@link List} of {@link Funcotation} for the given {@code variant}, {@code referenceContext}, and {@code featureContext}.
+     * @param variant {@link VariantContext} to annotate.
+     * @param referenceContext {@link ReferenceContext} corresponding to the given {@code variant}.
+     * @param featureList {@link List} of {@link Feature} corresponding to the given {@code variant}.
```

Comment:
`featureList` is currently the set of features that matched with a given genomic region.  It originates from the Funcotator implementation of VariantWalker::apply.

It's intended to hold any features that might be applicable to a given variant so that they can be added as annotations to the output.